packages/malloy/src/lang/parse-to-ast.ts

```diff
@@ -900,6 +900,10 @@ export class MalloyToAST
     return new ast.ExprCountDistinct(this.getFieldExpr(pcx.fieldExpr()));
   }
 
+  visitExprUngroup(pcx: parse.ExprUngroupContext): ast.ExprUngroup {
+    return new ast.ExprUngroup(this.getFieldExpr(pcx.fieldExpr()));
```

Comment:
If an AST object generates errors which need to be scoped to the object, this gesture copes the text extend from the parse node into the AST so that error reporting works.

If you don't do that, it walks up the AST until it finds a parent that as a known location.

```Typescript
return this.astAt(pcx, new ast.ExprUngroup(this.getFieldExpr(pcx.fieldExpr())));
```

Reply:
Yep, thanks. Totally forgot about doing that. The `visitExprCountDistinct` (directly preceding) should have that too, right?

Reply:
It doesn't hurt to put it everywhere.
If the node doesn't error report relative to itself, it isn't needed.